File: src/main/java/com/polygloat/dtos/request/validators/ValidationError.java
package com.polygloat.dtos.request.validators;

import com.polygloat.constants.Message;
import lombok.Getter;

public class ValidationError {
    @Getter
    private String[] parameters;

    @Getter
    private ValidationErrorType type;

    @Getter
    private Message message;

    public ValidationError(ValidationErrorType type, Message message, String... parameters) {
        this.parameters = parameters;
        this.message = message;
        this.type = type;
    }
}

File: backend/core/src/main/kotlin/io/tolgee/dtos/request/validators/ValidationError.kt
package io.tolgee.dtos.request.validators

import io.tolgee.constants.Message

class ValidationError(
  type: ValidationErrorType,
  message: Message,
  vararg parameters: String,
) {
  val parameters: Array<String> = parameters as Array<String>
  val type: ValidationErrorType = type
  val message: Message = message
}
